homeassistant/components/velbus/sensor.py

```diff
@@ -35,32 +42,54 @@ def unique_id(self):
             unique_id = f"{unique_id}-counter"
         return unique_id
 
+    @property
+    def name(self):
+        """Return the name for the sensor."""
+        name = super().name
+        if self._is_counter:
+            name = f"{name}-counter"
+        return name
+
     @property
     def device_class(self):
         """Return the device class of the sensor."""
-        if self._module.get_class(self._channel) == "counter" and not self._is_counter:
-            if self._module.get_counter_unit(self._channel) == ENERGY_KILO_WATT_HOUR:
-                return DEVICE_CLASS_POWER
-            return None
-        return self._module.get_class(self._channel)
+        if self._is_counter:
+            return DEVICE_CLASS_ENERGY
+        if self._channel.is_counter_channel():
+            return DEVICE_CLASS_POWER
+        if self._channel.is_temperature():
+            return DEVICE_CLASS_TEMPERATURE
+        return None
 
     @property
     def state(self):
         """Return the state of the sensor."""
         if self._is_counter:
-            return self._module.get_counter_state(self._channel)
-        return self._module.get_state(self._channel)
+            return self._channel.get_counter_state()
+        return self._channel.get_state()
 
     @property
     def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
         if self._is_counter:
-            return self._module.get_counter_unit(self._channel)
-        return self._module.get_unit(self._channel)
+            return self._channel.get_counter_unit()
+        return self._channel.get_unit()
 
     @property
     def icon(self):
         """Icon to use in the frontend."""
         if self._is_counter:
             return "mdi:counter"
         return None
+
+    @property
+    def state_class(self):
+        """Return the state class of this device."""
+        return STATE_CLASS_MEASUREMENT
+
+    @property
+    def last_reset(self) -> datetime | None:
+        """Return the last reset date of this sensor."""
+        if self._is_counter:
+            return utc_from_timestamp(0)
```

Comment:
Not sure if this is right. In the end it's possible to (manually) reset the counter outside of HomeAsssitant. So the `last_reset` timestamp should somehow reflect that moment of reset. Although I realize the velbus devices does expose the actual timestamp of the last reset.

However based on recent merge of #54523 I understand we don't need last_reset at all. Just change the return of `state_class` into `STATE_CLASS_TOTAL_INCREASING` and drop the this `last_reset `property.

Reply:
Indeed, the PR was added after I added this code.

So will change it after my holidays